src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```diff
@@ -65,25 +59,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Enable the trim analyzer when any trimming settings are enabled. Warnings may suppressed based on other settings. -->
     <EnableTrimAnalyzer Condition="'$(EnableTrimAnalyzer)' == '' And ('$(PublishTrimmed)' == 'true' Or '$(IsTrimmable)' == 'true')">true</EnableTrimAnalyzer>
 
-    <!-- Compiler warning level, defaulted to 4. We promote it to 5 if the user has set analysis level to 5 or higher
-         NOTE: at this time only the C# compiler supports warning waves -->
-    <WarningLevel Condition="'$(Language)' == 'C#'">5</WarningLevel>
-  </PropertyGroup>
-
-  <!-- Set the default WarningLevel based on EffectiveAnalysisLevel
-       For .NET 6 we want the Warning level to be 6 -->
-  <PropertyGroup Condition="'$(EffectiveAnalysisLevel)' != '' And
-                             $([MSBuild]::VersionGreaterThanOrEquals($(EffectiveAnalysisLevel), '6.0'))">
-    <WarningLevel Condition="'$(Language)' == 'C#'">6</WarningLevel>
-  </PropertyGroup>
+    <!-- If the user specified 'preview' we want to pick a very high warning level to opt into the highest possible warning wave -->
+    <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(AnalysisLevel)' == 'preview'">9999</WarningLevel>
+    <!-- NOTE: at this time only the C# compiler supports warning waves like this. -->
+    <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == ''">$(_TargetFrameworkVersionWithoutV.Substring(0, 1))</WarningLevel>
```

Comment:
Would the second `WarningLevel` always overwrite the first one?

Reply:
This second one shouldn't be hit because the WarningLevel should be 9999 at this point, so the second condition should fail. I can definitely add test cases to cover this scenario, though. "When analysis level is preview warning level is 9999".

Reply:
> This second one shouldn't be hit because the WarningLevel should be 9999 at this point, so the second condition should fail.

Oh yes. My bad!